core/domain/exp_services.py

```diff
@@ -586,6 +588,277 @@ def apply_change_list(exploration_id, change_list):
+    change_list, old_metadata, new_metadata, current_version, committer_id):
+    """Updates the version history of the exploration at a particular version
+    of an exploration.
+
+    Args:
+        metadata_last_edited_version_number: int. Version number of the
+            exploration on which the metadata was last updated.
+        metadata_last_edited_committer_id: str. User id of the user who
+            committed the last change in the exploration metadata.
+        change_list: list(ExplorationChange). A list of changes introduced in
+            this commit.
+        old_metadata: ExplorationMetadata. The exploration metadata at the
+            previous version of the exploration.
+        new_metadata: dict(str, State). The exploration metadata at the
+            current version of the exploration.
+        current_version: int. The latest version of the exploration.
+        committer_id: str. The id of the user who made the commit.
+
+    Returns:
+        tuple[int, str]. The updated version number and the committer id.
+    """
+    old_metadata_dict = copy.deepcopy(old_metadata.to_dict())
+    new_metadata_dict = copy.deepcopy(new_metadata.to_dict())
+    prev_version = current_version - 1
+
+    metadata_was_changed = False
+    for change in change_list:
+        if change.cmd == exp_domain.CMD_EDIT_EXPLORATION_PROPERTY:
+            metadata_was_changed = True
+            break
+
+    if metadata_was_changed:
+        # The purpose of checking the diff_dict between the two metadata
+        # dicts ensure that we do not change the version history if the
+        # overall changes (by EDIT_EXPLORATION_PROPERTY) get cancelled by
+        # each other and there is no 'net change'.
+        diff_dict = deepdiff.DeepDiff(old_metadata_dict, new_metadata_dict)
+        if diff_dict != {}:
+            metadata_last_edited_version_number = prev_version
+            metadata_last_edited_committer_id = committer_id
+
+    return (
+        metadata_last_edited_version_number,
+        metadata_last_edited_committer_id)
+
+
+def get_updated_committer_ids(
+    states_version_history, metadata_last_edited_committer_id):
+    """Extracts a list of user ids who made the 'previous commit' on each state
+    and the exploration metadata from the exploration states and metadata
+    version history data.
+
+    Args:
+        states_version_history: dict(str, StateVersionHistory). The version
+            history data of each state at a particular version of an
+            exploration.
+        metadata_last_edited_committer_id: str. User id of the user who
+            committed the last change in the exploration metadata.
+
+    Returns:
+        list[str]. A list of user ids who made the 'previous commit' on each
+        state and the exploration metadata.
+    """
+    committer_ids = set()
+    for version_history in states_version_history.values():
+        committer_ids.add(version_history.committer_id)
```

Comment:
```suggestion
    committer_ids = set(states_version_history.values())
```

Reply:
Changed this as follows:
```
    committer_ids = set([
        version_history.committer_id
        for version_history in states_version_history.values()
    ])
```
Because we need the version_history.committer_id and not the whole dict.